packages/backend/src/ee/services/ai/models/openai-gpt.ts

```diff
@@ -10,34 +10,34 @@ export const getOpenaiGptmodel = (
         LightdashConfig['ai']['copilot']['providers']['openai']
     >,
     preset: ModelPreset<'openai'>,
+    options?: { enableReasoning?: boolean },
 ): AiModel<typeof PROVIDER> => {
     const openai = createOpenAI({
         apiKey: config.apiKey,
         ...(config.baseUrl ? { baseURL: config.baseUrl } : {}),
     });
+    const { supportsReasoning, modelId } = preset;
 
-    const model = openai(preset.modelId);
+    const model = openai(modelId);
 
-    const isGpt5 = model.modelId.includes('gpt-5');
-
-    const reasoningEnabled = preset.supportsReasoning;
+    const reasoningEnabled = options?.enableReasoning;
 
     return {
         model,
         callOptions: {
-            ...(!isGpt5 && {
-                // gpt-5 models don't support temperature
-                temperature: preset.callOptions.temperature,
-            }),
+            ...preset.callOptions,
+            // temperature is not supported when reasoning is enabled
+            ...(options?.enableReasoning
+                ? { temperature: 0.2 }
+                : { temperature: undefined }),
```

Comment:
this needs to be swapped like the other models ^

Reply:
fixed in upstack